### PR TITLE
[WIP] Remove deprecated KubeException

### PR DESCRIPTION
**Pull request by moolitayer**
See discussion in https://github.com/abonas/kubeclient/pull/195

Part of: https://github.com/abonas/kubeclient/issues/199

### CODE REVIEW

**Comment by moolitayer** [comment]
@cben @grosser @kirs please review

**Comment by moolitayer** [comment]
@jhernand please take a look

File: README.md [hunk 1]
```diff
@@ -476,6 +476,10 @@ client.process_template template
 
 ## Upgrading
 
+### past version 3.0
+
+Deprecated KubeException has been dropped, use `Kubeclient::HttpException` instead.
+
```
Comment (jhernand) [inline README.md]: Should be `Kubeclient::HttpError`.

**Comment by jhernand** [comment]
@pkliczewski, @masayag the KubeVirt provider uses the deprecated exception:

https://github.com/ManageIQ/manageiq-providers-kubevirt/blob/d46aca33eb6330b4e79a16100c6379e53a912ab5/app/models/manageiq/providers/kubevirt/infra_manager/refresh_worker/runner.rb#L192

Makes sure to change that to `Kubeclient::HttpError` before upgrading to the upcoming version 3.0 of `kubeclient`. You can change it now, if you want, as version 2.4 already supports it.

File: README.md (hunk 1 above)
Comment (moolitayer) [reply README.md]: see https://github.com/abonas/kubeclient/pull/288

**Comment by moolitayer** [comment]
@cben please review

**Comment by moolitayer** [comment]
Discussed with @cben 
Planning to release maser, as we haven't yet released a version where this is deprecated we should wait with this one.

**Comment by cben** [comment]
status: I've been postponing this as I see an increasing needs for errors where HttpError would make no sense, specifically around `Config`.  Plus if we'll get auth auto-renewal (#393), we might start seeing Config-related errors from `Client` methods (get_pods etc) that currently only raise HttpError.

All this leads to me to think kubeclient still needs a top class/module covering all its errors.
For now that's  `KubeException` but we might want to transition to say `Kubeclient::Error`.
As `Config` errors are currently various RuntimeError, KeyError etc, I want to figure out if it's reasonable to substitute a different class...
=> So, still stalling on this... :construction:

**Pull request closed without merging**